### PR TITLE
Revamp docs & codebase

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Revamp docs to speed up onboarding and a add some automated scripts

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [X] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work.
- [X] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/scripts/deploy_kind_cluster.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```
Comment (alexcreasy) [inline clients/ui/scripts/deploy_kind_cluster.sh:1]: Depending on the system, it's not guaranteed bash will be there, I'd recommend the shebang below, it's pretty much universal.

```suggestion
#!/usr/bin/env bash
```

**Review by ederign — APPROVED** [review]
Lucas great work! Worked like magic:

Screenshot 2024-11-08 at 2 22 31 PM


Suggested future improvements
- on make dev-bff: be able to configure the port and mock options
- small typo on deploykindcluster.sh L 53
- create a way to delete deployment of a cluster and rebuild and redeploy.

But all of those can be done in FUP PR's. I'll merge this because I'll make a FUP PR on top of it!.

**Review by rareddy — APPROVED** [review]
/lgtm
/approve

**Pull request merged**